nixos/modules/config/networking.nix

```diff
@@ -8,11 +8,12 @@ let
 
   cfg = config.networking;
 
-  localhostMapped4 = cfg.hosts ? "127.0.0.1" && elem "localhost" cfg.hosts."127.0.0.1";
-  localhostMapped6 = cfg.hosts ? "::1"       && elem "localhost" cfg.hosts."::1";
-
   localhostMultiple = any (elem "localhost") (attrValues (removeAttrs cfg.hosts [ "127.0.0.1" "::1" ]));
 
+  localhostHostname = # Note: The FQDN (canonical hostname) has to come first:
```

Comment:
I personally find `localhostHostname` very confusing. There's no `localhost` inside here. Should this be `localHostname`?

Reply:
Yes, I agree that it's confusing, I tried to come up with a better name but failed :o
It's basically a list/value for the required `/etc/hosts` entries: `[ $FQDN $HOSTNAME ]`.

Maybe I should also move this to `networking.hosts` (using `let ... in ...` there) and call it e.g. `hostnames`.

Reply:
I've added a possible improvement.